app/services/record_declaration.rb

```diff
@@ -182,9 +186,33 @@ def validates_billable_slot_available
     errors.add(:base, I18n.t(:declaration_already_exists))
   end
 
-  def participant_declaration_class_for(participant_profile)
-    return ParticipantDeclaration::NPQ if participant_profile.is_a?(ParticipantProfile::NPQ)
+  def participant_declaration_class
+    if participant_profile.npq?
+      ParticipantDeclaration::NPQ
+    else
+      ParticipantDeclaration::ECF
+    end
+  end
+
+  def validate_has_passed?
+    return false unless FeatureFlag.active?(:participant_outcomes_feature)
+
+    participant_profile&.npq? && declaration_type == "completed"
```

Comment:
could we also ignore ehco and aso NPQ declarations?

Reply:
@cwrw 
The `NPQ::CreateParticipantOutcome` service class won't create outcome if ehco and aso declarations.
Tom didn't want a validation error here, just not create outcome silently.

Reply:
I think however  we will still be validating has passed is required on those declarations, and might need to add an extra check here to avoid that validation?